tests/components/homewizard/test_sensor.py

```diff
@@ -637,3 +643,103 @@ async def test_sensor_entity_export_disabled_when_unused(
     )
     assert entry
     assert entry.disabled
+
+
+async def test_sensors_unreachable(hass, mock_config_entry_data, mock_config_entry):
+    """Test sensor handles api unreachable."""
+
+    api = get_mock_device()
+    api.data.available_datapoints = [
+        "total_power_import_t1_kwh",
+    ]
+    api.data.total_power_import_t1_kwh = 1234.123
+
+    with patch(
+        "aiohwenergy.HomeWizardEnergy",
+        return_value=api,
+    ):
+        api.update = AsyncMock(return_value=True)
+
+        entry = mock_config_entry
+        entry.data = mock_config_entry_data
+        entry.add_to_hass(hass)
+
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+        assert (
+            hass.states.get(
+                "sensor.product_name_aabbccddeeff_total_power_import_t1"
+            ).state
+            == "1234.123"
+        )
+
+        api.update = AsyncMock(return_value=False)
+        async_fire_time_changed(hass, utcnow() + timedelta(seconds=5))
```

Comment:
Probably safer to call utcnow first and store the result in a variable and then use that to generate different timepoijts together with a timedelta.